ibis/udf/vectorized.py

```diff
@@ -198,7 +207,7 @@ def _get_coercion_function(self):
                 return _coerce_to_dataframe
             else:
                 # Case 2: Struct output, reduction UDF -> coerce to tuple
-                return _coerce_to_tuple
+                return _coerce_to_dict
```

Comment:
Otherwise [ops.StructField implementation](https://github.com/ibis-project/ibis/pull/3497/files#diff-0eb3b553dfe84f503ecd43188a21a2df3aa106437939772d608c7c13f09722bcR15) would have to deal with both integer and string indices.